simtools/db_handler.py

```diff
@@ -1262,3 +1283,60 @@ def insertFilesToDB(self, filesToAddToDB, dbName=DB_CTA_SIMULATION_MODEL):
             self.insertFileToDB(fileNow, dbName, **kwargs)
 
         return
+
+    def getAllVersions(
+        self,
+        dbName,
+        parameter,
+        telescopeModelName=None,
+        site=None,
+        collectionName="telescopes",
+    ):
+        """
+        Get all version entries in the DB of a telescope or site for a specific parameter.
+
+        Parameters
+        ----------
+        dbName: str
+            the name of the DB
+        parameter: str
+            Which parameter to get the versions of
+        telescopeModelName: str
+            Which telescope to get the versions of (in case "collectionName" is "telescopes")
+        site: str, North or South
+            In case "collectionName" is "telescopes", the site is used to build the telescope name.
+            In case "collectionName" is "sites",
+            this argument sets which site parameter get the versions of
+        collectionName: str
+            The name of the collection in which to update the parameter (default is "telescopes")
+
+        Returns
+        -------
+        allVersions: list
+            List of all versions found
+        """
+
+        collection = DatabaseHandler.dbClient[dbName][collectionName]
+
+        query = {
+            "Parameter": parameter,
+        }
+
+        _siteValidated = names.validateSiteName(site)
+        if collectionName == "telescopes":
+            _telModelNameValidated = names.validateTelescopeModelName(telescopeModelName)
+            _telNameDB = self._getTelescopeModelNameForDB(_siteValidated, _telModelNameValidated)
+            query["Telescope"] = _telNameDB
+        elif collectionName == "sites":
+            query["Site"] = _siteValidated
+        else:
+            raise ValueError("Can only get versions of the telescopes and sites collections.")
+
+        _allVersions = list()
+        for post in collection.find(query):
```

Comment:
Ha! Suggest to replace the non-pythonic for loop by list comprehension...:-)

Reply:
When you're right, you're right!